app/models/miq_worker.rb

```diff
@@ -128,8 +128,30 @@ def self.enough_resource_to_start_worker?
     MiqServer.my_server.enough_resource_to_start_worker?(self)
   end
 
+  def self.cleanup_failed_workers
+    if containerized_worker?
+    elsif systemd_worker?
+      failed_services.each do |service|
+        cleanup_systemd_worker(service)
+      end
+    end
+  end
+
+  def self.current_workers
+    if containerized_worker?
+      # TODO search container deployments
+      include_stopping_workers_on_synchronize ? find_alive : find_current_or_starting
+    elsif systemd_worker?
+      active_services
+    else
+      include_stopping_workers_on_synchronize ? find_alive : find_current_or_starting
```

Comment:
So, from a higher architectural level, I wonder if we can have the core server worker monitor code and core worker code and have systemd/appliance process/containers mixins that get included based on what style we're using.  We can then eliminate lots of these `containerized_worker?`, `systemd_worker?` checks and just ensure each implement the required methods on the server and worker classes.

In other words, do something like this:

```ruby
class MiqWorker < ApplicationRecord
  include_concern 'ContainerCommon' if MiqEnvironment::Command.is_podified? && supports_container?
  include_concern 'SystemdCommon'   if MiqEnvironment::Command.supports_systemd? && supports_systemd?
                                    # ^^^ the body of containerized_worker? and systemd_worker?
```
But it looks like supports_container? and supports_systemd? might be specific for some workers so there might be reasons it wasn't done that way and is instead done like this:

```ruby
class MiqWorker < ApplicationRecord
  include_concern 'ContainerCommon'
  include_concern 'SystemdCommon'
```

Reply:
I was thinking more along the lines of a 

```
class WorkerMangement::SystemdManager
  def current_workers
  end
end

class WorkerManagement::KubernetesManager
  def current_worker
  end
end
```

Then somewhere we just have a single switch that picks the right manager.  We could also have managers for the current spawn-based code (which we still may need for things like the monolithic docker image), and possibly even a foreman manager if someone wanted to spin up an orchestrator locally

Reply:
yeah, delegating to the right manager would be nice to avoid any conditionals at this level.

Reply:
This is good conversation and we'll tackle this as we get down to the "current_workers"... for now, this PR just handles the failed services so I think we can resolve this.